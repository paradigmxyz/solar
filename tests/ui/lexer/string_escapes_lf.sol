// Escaped - OK
string constant s = "\
";
string constant s = unicode"\
";
bytes constant s = hex"\
";
//~^^ ERROR: invalid hex digit
//~| ERROR: invalid hex digit
// 2 for \\, \n

// Escaped, but can only escape one newline
string constant s = "\

"; //~^ ERROR: cannot skip multiple lines
string constant s = unicode"\

"; //~^ ERROR: cannot skip multiple lines
bytes constant s = hex"\

";
//~^^^ ERROR: invalid hex digit
//~| ERROR: invalid hex digit
//~^^^^ ERROR: invalid hex digit
// 3x for \\, \n, \n

// Unescaped
string constant s = "
"; //~^ ERROR: unescaped newline
string constant s = unicode"
"; //~^ ERROR: unescaped newline
bytes constant s = hex"
";
//~^^ ERROR: invalid hex digit
// 1x for \n
