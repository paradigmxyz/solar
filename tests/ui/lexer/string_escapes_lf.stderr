error: invalid hex digit
  --> ROOT/tests/ui/lexer/string_escapes_lf.sol:LL:CC
   |
LL | bytes constant s = hex"\
   |                        ^
   |

error: invalid hex digit
  --> ROOT/tests/ui/lexer/string_escapes_lf.sol:LL:CC
   |
LL | bytes constant s = hex"\
   |                         ^
   |

error: cannot skip multiple lines with `\`
  --> ROOT/tests/ui/lexer/string_escapes_lf.sol:LL:CC
   |
LL |
   | ^
   |

error: cannot skip multiple lines with `\`
  --> ROOT/tests/ui/lexer/string_escapes_lf.sol:LL:CC
   |
LL |
   | ^
   |

error: invalid hex digit
  --> ROOT/tests/ui/lexer/string_escapes_lf.sol:LL:CC
   |
LL | bytes constant s = hex"\
   |                        ^
   |

error: invalid hex digit
  --> ROOT/tests/ui/lexer/string_escapes_lf.sol:LL:CC
   |
LL | bytes constant s = hex"\
   |                         ^
   |

error: invalid hex digit
  --> ROOT/tests/ui/lexer/string_escapes_lf.sol:LL:CC
   |
LL |
   | ^
   |

error: unescaped newline
  --> ROOT/tests/ui/lexer/string_escapes_lf.sol:LL:CC
   |
LL | string constant s = "
   |                      ^
   |

error: unescaped newline
  --> ROOT/tests/ui/lexer/string_escapes_lf.sol:LL:CC
   |
LL | string constant s = unicode"
   |                             ^
   |

error: invalid hex digit
  --> ROOT/tests/ui/lexer/string_escapes_lf.sol:LL:CC
   |
LL | bytes constant s = hex"
   |                        ^
   |

error: aborting due to 10 previous errors

